include/rcpputils/filesystem_helper.hpp

```diff
@@ -183,16 +183,31 @@ inline bool create_directories(const path & p)
 {
   path p_built;
 
-  for (auto it = p.cbegin(); it != p.cend(); ++it) {
-    p_built /= *it;
-
+  int status = 0;
+  for (auto it = p.cbegin(); it != p.cend() && status == 0; ++it) {
+    if (p_built.empty()) {
+      p_built = *it;
```

Comment:
This just happens in the first iteration, right?
Consider doing it before the loop (though I don't mind much).

Reply:
Yeap, but then I'm forced to copy loop code. Tried to make it not so, but it gets ugly. Compiler will likely do loop unrolling anyways.